### PR TITLE
AP_HAL_SITL clean 1

### PR DESCRIPTION
**Pull request by khancyr**
Taken from https://github.com/ArduPilot/ardupilot/pull/5639

 finish fixing some style issue : catch with cpplint, cppcheck, and clion ide
- rework include from 
[master.pdf](https://github.com/ArduPilot/ardupilot/files/736023/master.pdf)
 to 
[clean.pdf](https://github.com/ArduPilot/ardupilot/files/736024/clean.pdf) : tools used : clion ide, cpplint, https://www.flourish.org/cinclude2dot/ (only include in ap_hal_sitl are show)
- correct some variable, param, function 
- make some variable const

### CODE REVIEW

**Comment by khancyr** [comment]
rebased on master

File: libraries/AP_HAL_SITL/SITL_State.cpp [hunk 1]
```diff
@@ -327,10 +316,8 @@ void SITL_State::_simulator_servos(SITL::Aircraft::sitl_input &input)
     /* this maps the registers used for PWM outputs. The RC
      * driver updates these whenever it wants the channel output
      * to change */
-    uint8_t i;
-
     if (last_update_usec == 0) {
-        for (i=0; i<SITL_NUM_CHANNELS; i++) {
+        for (uint8_t i = 0; i < SITL_RC_OUTPUT_CHANNELS; i++) {
```
Comment (nicolaerosia) [inline libraries/AP_HAL_SITL/SITL_State.cpp]: int should be faster since we're running on 32bit processors

**Comment by khancyr** [comment]
rebased

**Comment by khancyr** [comment]
closed as too much time pass, I will rework and reopen later

**Pull request closed without merging**